File: TransCoder_tokenized_test_set_functions/python/FIND_FOUR_ELEMENTS_A_B_C_AND_D_IN_AN_ARRAY_SUCH_THAT_AB_CD.py
def findPairs ( arr, n ) :
Hash = {
}

for i in range ( n - 1 ) :
    for j in range ( i + 1, n ) :
    sum = arr [ i ] + arr [ j ]
    if sum in Hash . keys ( ) :
        prev = Hash . get ( sum )
        print ( str ( prev ) + " and (%d, %d)" % ( arr [ i ], arr [ j ] ) )
        return True
    else :
        Hash [ sum ] = ( arr [ i ], arr [ j ] )
    

File: TransCoder_tokenized_test_set_functions/cpp/FIND_FOUR_ELEMENTS_A_B_C_AND_D_IN_AN_ARRAY_SUCH_THAT_AB_CD.cpp
bool findPairs ( int arr [ ], int n ) {
  map < int, pair < int, int > > Hash;
  for ( int i = 0;
  i < n;
  ++ i ) {
    for ( int j = i + 1;
    j < n;
    ++ j ) {
      int sum = arr [ i ] + arr [ j ];
      if ( Hash . find ( sum ) == Hash . end ( ) ) Hash [ sum ] = make_pair ( i, j );
      else {
        pair < int, int > pp = Hash [ sum ];
        cout << "(" << arr [ pp . first ] << ", " << arr [ pp . second ] << ") and (" << arr [ i ] << ", " << arr [ j ] << ")n";
        return true;
      }
    }
  }
  cout << "No pairs found";
  return false;
}